connectors/services/sync.py

```diff
@@ -45,8 +45,19 @@ def __init__(self, config, args):
             "max_concurrent_syncs", DEFAULT_MAX_CONCURRENT_SYNCS
         )
         self.connectors = None
+        self.syncs = ConcurrentTasks(max_concurrency=self.concurrent_syncs)
```

Comment:
we can't have a single instance here because when it's joined it's thrown away - if we want to reuse the same instance of `ConcurrentTasks` you need to change the class so `join` empties `self.tasks`

Reply:
there's also one problem here. if the same instance is used by competing tasks to put stuff, join() will take into account *all* the tasks -- I think it would be better to put this back in the loop (unless you had a requirement to make it a single instance)

Reply:
I think I had the similar discussion with @tarekziade , and here are my inputs:

1. I agree we should assign ConcurrentTasks to an instance variable, so we could access it when needed
2. Given point 1), the instantiation should happen outside the while loop (`while self.running`)
3. I believe we should put [connector.sync()](https://github.com/elastic/connectors-python/blob/0004f6badfae91527febfa9c837da96c2184fee5/connectors/services/sync.py#L88) inside the `ConcurrentTasks` (Later it will be [sync_job_runner.execute()](https://github.com/elastic/connectors-python/blob/014a189a947cba1e6c70b5d212feebdbf937eb81/connectors/services/sync.py#L101)), instead of `_one_sync()`
4. I don't quite understand why we need to use `await self.syncs.join()`. It will block the while loop into the next iteration until all tasks are done. The task will be scheduled to run in the background once `put`, so we don't really care when it's done.